### ISSUE TITLE
[🐞] JS Load Regression

### ISSUE THREAD

**Issue by zanettin** [BuilderIO/qwik#3390]
### Which component is affected?

Qwik Runtime

### Describe the bug

Versions > 0.18.1 are loading about 25k of JS which should not be needed from the app perspective.

# Reproduction
## "Good" version
```
pnpm create qwik@0.18.1
cd qwik-app
```
open `src/routes/index.tsx` in the editor and remove the usage of `<Link>` component
```
pnpm run preview
```
open dev tools -> network and check js downloads. it should be 0.

## "Optimizable" version
do the same as above but use latest version of qwik (0.21.0).
on my end it results in ~25k of JS.

# Screenshots v0.21.0
![Bildschirmfoto 2023-03-15 um 21 27 00](https://user-images.githubusercontent.com/3241476/225434243-02f8d323-a2c3-461b-945d-63a18828b013.png)
(network view)

![Bildschirmfoto_2023-03-15_um_20_57_12](https://user-images.githubusercontent.com/3241476/225434245-74e800e1-75da-44e7-9c0d-d9dd7c56c2e3.png)
(this line causes the download of the first q-xxxxx pkg)

### Reproduction

none

### Steps to reproduce

pls see above

### System Info

```shell
not relevant in this case
```

### Additional Information

Seems identical to what is described here 
- https://github.com/BuilderIO/qwik/issues/3298
- https://github.com/BuilderIO/qwik/issues/2998

**Comment by Coretteket** [comment]
Thanks for opening this issue, @zanettin! I'm just wondering if there's any significance in changing the milestone from Priority to Planned? I really appreciate Qwik's commitment to an initial load of 0kb JS, so an initial load of 25kb is quite a turn-off in my view. Thanks again for all the work on this project!

**Comment by cmbartschat** [comment]
I took a look at this and narrowed the regression to the qwik-city release 0.5.3 (March 7) being the first to display that issue.

I'm guessing the change to either [submodule-build.ts](https://github.com/BuilderIO/qwik/commit/8db17e40f2515708cf76ca368754a95e0a40a136#diff-d7abe7579e1ebc03fa6f095f564d8f42a05ef9988c0c6c4aa27dc1bee272bcd0L29-L34) or [submodule-core.ts](https://github.com/BuilderIO/qwik/commit/8db17e40f2515708cf76ca368754a95e0a40a136#diff-d603198078887d81d5e294ddcd59d34ce79f5995b0843be1b98f759d45993c3bL61-R73) is the culprit.

@manucorporat do we understand if those changes could have caused this?

**Comment by Dindaleon** [comment]
I am waiting for this to be fixed soon as I halted all development on my project because this is an important feature for it. I would like to know if this is going to be addressed within the upcoming releases.

**Comment by Varixo** [comment]
So the first, larger file is from vite:
https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/importAnalysisBuild.ts

The second file is from:
https://github.com/BuilderIO/qwik/blob/63bd095ad5575db0ffb642f6f356a4e621ce49de/packages/qwik-city/runtime/src/router-outlet-component.ts#L21

**Comment by manucorporat** [comment]
This got finally fixed in 1.1.0!
